rules/rules-overridden-azure/cloud-readiness/mail.windup.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset detects the Java Mail API, which may be problematic when migrating an application to a cloud environment.
+            It overrids rules/rules-reviewed/cloud-readiness/mail.windup.xml
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <targetTechnology id="azure-spring-apps"/>
+        <targetTechnology id="azure-appservice"/>
+        <targetTechnology id="azure-aks"/>
+        <targetTechnology id="azure-container-apps"/>
+        <tag>mail</tag>
+        <overrideRules>true</overrideRules>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                <or>
+                    <javaclass references="javax.mail.{*}"/>
+                    <javaclass references="com.sun.mail.{*}"/>
+                    <javaclass references="org.simplejavamail.{*}"/>
+                    <javaclass references="org.apache.commons.mail.{*}"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Java Mail API" effort="3" category-id="potential">
+                    <message>
+                        The application uses email APIs to send or receive emails.
+                        Cconsider using Azure Communication Services Email to migrate the email functionality of your application to Azure.
+
+                        * Set up **Azure Communication Services**: Create an Azure Communication Services resource in your Azure subscription. This will provide you with the necessary capabilities to send emails.
+
+                        * **Configure email sending**: Update your application's email sending code to use the Azure Communication Services Email SDK or REST API. This involves integrating the Azure Communication Services Email client libraries into your application and configuring the required settings, such as the sender's email address, recipient addresses, subject, and email content.
+
+                        * **Manage email templates**: Azure Communication Services Email allows you to create and manage email templates. Consider leveraging this feature to create reusable templates for consistent email formatting and content.
+                        
+                        **WARNING** Check if port 25 is open in the App Service SKU.
```

Comment:
If the customer is utilizing ACS to send emails, do they still need Port 25 open? This may be a case of a rule that provides two solutions:

1. Use ACS
2. Use SMTP directly, but "here's what you must do to make it work"

Reply:
I think the rule should focus on config and backend service for cloud readiness but not migrate to ACS.  For my point of view, the mail service is different with database as few customers will migrate the mail service. 

Can we keep the message as below and left less ACS links?
```
<message>
                    The configuration parameters of SMTP services for outbound mail may contain environment-specific values such as static IP addresses, host names, properties, credentials, or certificates.

                    Recommendations

                    * In a cloud environment, mail systems should be considered backing services and treated like attached resources.
                    * Ensure that the configuration of the underlying outbound mail connection is not environment-specific.
                    * In OpenShift Container Platform, store configuration parameters as environment variables or in a ConfigMap.
</message>
                <link href="https://12factor.net/backing-services" title="Twelve-factor app - Backing services"/>
                <link href="https://12factor.net/config" title="Twelve-factor app - Config"/>
```

Reply:
I agree with @showpune here. Customer will need less effort if they don't have to migrate to ACS.